assets/src/edit-story/components/inspector/prepublish/prepublishChecklistProvider.js

```diff
@@ -61,9 +66,28 @@ function PrepublishChecklistProvider({ children }) {
     }
   }, [handleRefreshList, refreshOnInitialLoad, refreshOnPageSizeChange]);
 
+  const [checkpointState, dispatch] = useReducer(
+    checkpointReducer,
+    PPC_CHECKPOINT_STATE.ONLY_RECOMMENDED
+  );
+
+  // Check for different qualifications to be met to update current PPC checkpoint
+  // 1. Story is no longer empty (ON_DIRTY_STORY)
+  // 2. Publish button is hit on a draft (ON_PUBLISH_CLICKED)
+  // 3. Story has more than 4 pages
+  useEffect(() => {
+    if (story.pages.length > 4) {
+      dispatch(PPC_CHECKPOINT_ACTION.ON_STORY_HAS_5_PAGES);
+    }
+  }, [story?.pages]);
```

Comment:
if the story becomes less than 4 pages, should we dispatch a similar action to clear this state?

Reply:
I think the nice thing with state machines is you don't really have to worry about how much this dispatch gets called. Also the PPC has like a one way progression modeled by the state machine (and requested by UX), so there's no need to stop computing errors or something once they've already been computed.

It's just nice that no matter how many times this fires, the state machine has the logic built in where we get the scenario of `only perform an action when the state is X and Y action fires`

Reply:
But to answer the question more directly, I don't think there's any UX requests that I know of that say to stop computing the errors in PPC once you've already started computing them.

Reply:
Love it - totally makes sense. That makes things much simpler